Automattic/wp-calypso
# Playwright: Create multiple jest suites

*** Note, there is an alternative approach taken in PR #56069. We should only merge one of these two. :) 

### Changes proposed in this Pull Request

Implements a new strategy for adding testing suites to our playwright tests. For background, see pciE2j-sX-p2.

Suites are created through a combination of spec file name suffixes and Jest config files that pick up those files.

Current suites:
- `calypso-pr`
- `calypso-release`
- `jetpack`
- `gutenberg` (FKA wpcom)

### Testing instructions

Locally, you can run jest with all of the given configs. It should pick up the right suites based on the name suffixes in the `specs/specs-playwright` folder.

- [ ] `yarn jest --config jest.config.calypos-pr.js` should pick up 18 suites.
- [ ] `yarn jest --config jest.config.gutenberg.js` should pick up 7 suites.
- [ ] `yarn jest --config jest.config.calypos-release.js` shouldn't pick up any yet.
- [ ] `yarn jest --config jest.config.jetpack.js` shouldn't pick up any yet.

We also need to make sure running individual specs works!

- [ ] `yarn jest specs/specs-playright/<pick-your-poison!>`

We can also see the TeamCity update for the Calypso PR build right here!

- [ ] The Playwright E2E builds should look like they're running the right tests. Did I pick the right specs?**

** Note, this will be a smaller number than before. Before, we were running all transitioned WPCOM tests as part of every PR! Now, we are only running tests that should be relevant to Calypso!

### Post-review To Do

- [ ] Update READMEs to match. (I prefer to get the first review done and make sure there's no big changes, then I'll push up the docs!)
